registry/modules/stripe-subscriptions/controllers/subscription.ts

```
@@ -0,0 +1,354 @@
+            plan: newPlanKey,
+            customerId: user.customerId,
+          });
+
+          updatedSubscriptions.push(userSub);
+        }
+
+        return {
+          plan: newPlanKey,
+          isOwner: true,
+          seats: newSubscriptionItem.quantity,
+          subscriptionId: updatedSubscriptions[0].subscriptionId,
+          createdAt: updatedSubscriptions[0].createdAt,
+        };
+      } else {
+        throw subscriptionNotFound(subscriptionId);
+      }
+    },
+
+    async updateSeats(props) {
+      const { userId, subscriptionId, newSeats } = props;
+
+      const subscription = await stripe.subscriptions.retrieve(subscriptionId);
+
+      if (parseInt(subscription.metadata.userId) === userId) {
+        // User is owner of this subscription
+        const subscriptionItem = subscription.items.data[0];
+        const subscriptionSeats = subscriptionItem.quantity!;
+
+        if (newSeats === subscriptionSeats) return;
+        else if (newSeats < subscriptionSeats) {
+          // Removing seats, check if there are available empty seats
+          const subscriptionUsers =
+            await userSubRepository.getSubscriptionUsers(subscriptionId);
+
+          if (newSeats < subscriptionUsers.length) {
+            // No available seats to remove
+            throw noEmptySeatsToRemove();
+          }
+        }
+
+        // Update seats
+        await stripe.subscriptionItems.update(subscriptionItem.id, {
+          quantity: newSeats,
+        });
+      } else {
+        throw subscriptionNotFound(subscriptionId);
+      }
+    },
+
+    async removeUserFromSeat({ userId, subscriptionId }) {
+      const userSub = await userSubRepository
+        .getUserSubscriptions(userId)
+        .then(
+          (data) => data.filter((el) => el.subscriptionId === subscriptionId)[0]
+        );
+
+      if (!userSub.isOwner) {
+        userSubRepository.removeUserFromSubscription(userId, subscriptionId);
```

await  userSubRepository.removeUserFromSubscription(userId, subscriptionId);

---

`userSubRepository.removeUserFromSubscription` doesn't return anything, so its not a Promise

---

It is still a promise even if it does not return anything. If you do not await it, if `removeUserFromSubscription` errors, that error will be hidden.